src/main/java/uk/gov/register/resources/DeleteRegisterDataResource.java

```diff
@@ -30,6 +32,7 @@ public DeleteRegisterDataResource(RegisterContext registerContext, ConfigManager
     public Response deleteRegisterData() throws NoSuchConfigException, IOException {
         flyway.clean();
         configManager.refreshConfig();
+        registerContext.resetMemoizationStore();
         flyway.migrate();
```

Comment:
I feel like this behaviour should be encapsulated into a `resetRegister()` method on `RegisterContext`? It's certainly not the resource's job to know that we need to reset both the db and the memoization store together.